open_spiel/games/chess/chess_board.cc

```diff
@@ -451,30 +454,36 @@ Square ChessBoard<kBoardSize>::find(const Piece &piece) const {
   return InvalidSquare();
 }
 
-template <uint32_t kBoardSize>
-void ChessBoard<kBoardSize>::GenerateLegalMoves(
+template <uint32_t kBoardSize, bool kAllowCheck>
+void ChessBoard<kBoardSize, kAllowCheck>::GenerateLegalMoves(
     const MoveYieldFn &yield) const {
-  auto king_square = find(Piece{to_play_, PieceType::kKing});
 
-  GeneratePseudoLegalMoves([this, &king_square, &yield](const Move &move) {
-    // See if the move is legal by applying, checking whether the king is
-    // under attack, and undoing the move.
-    // TODO: Optimize this.
-    auto board_copy = *this;
-    board_copy.ApplyMove(move);
+  if (kAllowCheck) {
+    GeneratePseudoLegalMoves(yield);
+  }
+  else {
+    auto king_square = find(Piece{to_play_, PieceType::kKing});
 
-    auto ks = at(move.from).type == PieceType::kKing ? move.to : king_square;
+    GeneratePseudoLegalMoves([this, &king_square, &yield](const Move &move) {
+      // See if the move is legal by applying, checking whether the king is
+      // under attack, and undoing the move.
+      // TODO: Optimize this.
+      auto board_copy = *this;
+      board_copy.ApplyMove(move);
 
-    if (board_copy.UnderAttack(ks, to_play_)) {
-      return true;
-    } else {
-      return yield(move);
-    }
-  });
+      auto ks = at(move.from).type == PieceType::kKing ? move.to : king_square;
+
+      if (board_copy.UnderAttack(ks, to_play_)) {
+        return true;
```

Comment:
void function returns value?

Reply:
?

Reply:
It's the yield function that returns boolean here. 

using MoveYieldFn = std::function<bool(const Move&)>;